hpx/runtime/serialization/access.hpp

```diff
@@ -36,6 +34,32 @@ namespace hpx { namespace serialization
 
     class access
     {
+        template <class T>
+        class has_serialize
+        {
+            struct two_chars
+            {
+                char _[2];
+            };
```

Comment:
Can we use `std::true_type` and `std::false_type` here?

Reply:
Not that I can think of.
Do you mean to use std::true/false_type as the return types for the test functions? If so, I don't think it makes a big difference. If you mean to inherit from std::true/false_type, then see the related discussion in #3134 why it didn't work for gcc.